packages/session-replay-browser/src/session-replay.ts

```diff
@@ -89,6 +119,19 @@ export class SessionReplay implements AmplitudeSessionReplay {
       ),
     );
 
+    if (options.sessionId && this.config.interactionConfig?.enabled) {
+      const transport = new BeaconTransport<ScrollEvent>(
+        {
+          sessionId: options.sessionId,
+          type: 'interaction',
+        },
+        this.config,
+      );
+      const scrollWatcher = new ScrollWatcher(transport);
```

Comment:
With a goal of keeping this top level class as simple as possible, ie only knowing what it needs to know about, wdyt about moving the instantiation of BeaconTransport to the constructor of ScrollWatcher? so that ScrollWatcher takes in the sessionId, type, and config, and calls `new BeaconTransport` within itself

Reply:
I'm thinking about doing this with another constructor. Being able to pass in the transport optionally makes it really easy to test.

Reply:
That works for me